ui/components/app/transaction-list/transaction-list.test.js

```diff
@@ -1,23 +1,102 @@
 import React from 'react';
-import { screen } from '@testing-library/react';
+import { fireEvent, screen } from '@testing-library/react';
 import { renderWithProvider } from '../../../../test/jest';
 import configureStore from '../../../store/store';
 import mockState from '../../../../test/data/mock-state.json';
+import { MOCK_ACCOUNT_BIP122_P2WPKH } from '../../../../test/data/mock-accounts';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventLinkType,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+import {
+  MULTICHAIN_NETWORK_BLOCK_EXPLORER_URL_MAP,
+  MULTICHAIN_TOKEN_IMAGE_MAP,
+  MultichainNetworks,
+} from '../../../../shared/constants/multichain/networks';
 import TransactionList from './transaction-list.component';
 
-const render = () => {
-  const store = configureStore({
-    metamask: {
-      ...mockState.metamask,
-      transactions: [],
+const defaultState = {
+  metamask: {
+    ...mockState.metamask,
+    transactions: [],
+  },
+};
+
+const btcState = {
+  metamask: {
+    ...mockState.metamask,
+    internalAccounts: {
+      ...mockState.metamask.internalAccounts,
+      accounts: {
+        ...mockState.metamask.internalAccounts.accounts,
+        [MOCK_ACCOUNT_BIP122_P2WPKH.id]: MOCK_ACCOUNT_BIP122_P2WPKH,
+      },
+      selectedAccount: MOCK_ACCOUNT_BIP122_P2WPKH.id,
     },
-  });
-  return renderWithProvider(<TransactionList />, store);
+    selectedAddress: MOCK_ACCOUNT_BIP122_P2WPKH.address,
+    providerConfig: {
+      chainId: MultichainNetworks.BITCOIN_TESTNET,
+      rpcUrl: '',
+      ticker: 'BTC',
+      nickname: 'Bitcoin (testnet)',
+      id: 'btc-testnet',
+      type: 'rpc',
+      rpcPrefs: {
+        imageUrl: MULTICHAIN_TOKEN_IMAGE_MAP[MultichainNetworks.BITCOIN],
+        blockExplorerUrl:
+          MULTICHAIN_NETWORK_BLOCK_EXPLORER_URL_MAP[
+            MultichainNetworks.BITCOIN_TESTNET
+          ],
+      },
+      isAddressCompatible: true,
+    },
```

Comment:
I don't think you need that, since the providers configs for non-EVM is an hardcoded list, could try re-run the test without this part?

Reply:
I tried without it but the provider config if hardcoded into the `mockState` as goerli so it will always return that unless you override it like we do here

Reply:
Then there might be something else wrong with the state, cause this `providerConfig` is EVM only.

If you try to debug around there: https://github.com/MetaMask/metamask-extension/blob/develop/ui/selectors/multichain.ts#L116-L131

You should see that we pass in the `isEvm` branch

I cannot re-test this myself, but I might give it a try tomorrow!

Reply:
Could you try to add `completedOnboarding: true,` to the state and remove the `providerConfig` again?

Reply:
Hey, setting `completedOnboarding: true` worked! I removed the `providerConfig` 😄 
[a4faff6](https://github.com/MetaMask/metamask-extension/pull/25642/commits/a4faff6dee05e13c751948962f9a984e9bc4c1d2)